### DISCUSSION TITLE
T41U5XBB USB to Ethernet

### DISCUSSION THREAD

**Discussion by raenji-sk** [phil-barrett/grblHAL-teensy-4.x#83] [General]
Hi, is it possible to change the connection on a T41U5XBB board from USB to Ethernet?

**Comment by phil-barrett** [comment]
If you use Windows to connect to the board, very easy. Use i[oSender](https://github.com/terjeio/ioSender/releases) and then follow [these instructions](https://www.grbl.org/single-post/networking-the-t41u5xbb).  If you do not have a router/switch, you can connect directly to a PC, use [these instructions.](https://www.grbl.org/single-post/ethernet-without-routers-for-grblhal)

If you use ioS or Linux, you can use the instructions for setting up grblHAL but you will need to use an emulator to create a Comm Port from a network connection for your GCode Sender to connect to.

**Comment by raenji-sk** [comment]
Thank you Phil, I wrote the question incorrectly. What I am looking for is if I can add the Ethernet capability to USB board. I do not have Ethernet connector on the board at the moment.  Soldering it to the board is no problem, but will it work like that or does it need some changes on the PCB (cut some traces, rewire pins, etc.)  At the time I ordered your board I preffered USB so that I could use more Gcode senders as most did not support ethernet connected controller. Now the situation is different and I all of my other controllers have ethernet. Not that I had many issues with USB but I find ethernet more stable and ressistant to electrical noise than USB. So if possible I would like to modify it this way. If it can be done can you please send me the product number or technical data sheet for the connector you're using so I can find one here locally?

**Comment by phil-barrett** [comment]
OK. No problem.  Yes, you can add Ethernet to your board. The [user manual](https://github.com/phil-barrett/grblHAL-teensy-4.x/blob/master/T41U5XBB%20v207.pdf) has the part numbers if you wish to order the parts yourself (near the end of the manual).  The magjack part number is very specific, only that part will work.  The 2mm socket is fairly easy to find but the 2mm header is extra tall and is harder to find.  Also, make sure you have do not have the Teensy 4.1-NE which omits the Ethernet Phy chip.  

I would be happy to sell you a kit but the shipping cost is the same as for the original board - I don't recall which country you live in but it is probably $19-20 USD on top of the $12 for the Ethernet kit. You might be able to do a lot better locally.

**Reply by raenji-sk** [reply]
That is great. Thank you. I am from Slovakia, so shipping might cost more than the parts. I'll look at the manual and try to get the parts locally. Worst case scenario I'll use it in USB mode for now and see how it goes.
